src/Avalonia.Controls.TreeDataGrid/Selection/TreeDataGridRowSelectionModel.cs

```diff
@@ -110,6 +113,99 @@ void ITreeDataGridSelectionInteraction.OnKeyDown(TreeDataGrid sender, KeyEventAr
 
         }
 
+        protected void HandleCharKey(string? text, TreeDataGrid treeDataGrid, int selectedRowIndex)
+        {
+            if (text != null && treeDataGrid.Columns != null)
+            {
+                var typedChar = text[0];
+
+                long now = DateTime.Now.Ticks;
+                long time = 0;
+                if (_lastCharPressedTime > 0)
+                {
+                    time = now - _lastCharPressedTime;
+                }
+
+                string candidatePattern;
+                if (time < 5000000)
+                {
+                    if (_typedWord.Length == 1 && typedChar == _typedWord[0])
+                    {
+                        candidatePattern = _typedWord;
+                    }
+                    else
+                    {
+                        candidatePattern = _typedWord + typedChar;
+                    }
+                }
+                else
+                {
+                    candidatePattern = typedChar.ToString();
+                }
+                foreach (var column in treeDataGrid.Columns)
+                {
+                    if (column is ITextSearchable textSearchableColumn && textSearchableColumn.IsTextSearchEnabled == true)
```

Comment:
Hello
We are from PlasticTeam, and while testing this feature we found a null exception here.
This is our fix:
 if (column is ITextSearchable textSearchableColumn && ((ITextSearchable)column).IsTextSearchEnabled == true)